### PR TITLE
Update dependencies and fix security issue

### PR DESCRIPTION
**Pull request by mmorel-35**
This also includes dependabot to handle security update for gomod and github-actions

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+version: 2
+updates:
+- package-ecosystem: gomod
+  directory: /
+  schedule:
+    interval: weekly
```
Comment (knqyf263) [inline .github/dependabot.yml:6]: We'll scan go dependencies by Trivy.

File: .github/workflows/bench.yml [hunk 2]
```diff
@@ -7,13 +7,13 @@ jobs:
     steps:
 
     - name: Set up Go
-      uses: actions/setup-go@v1
+      uses: actions/setup-go@v2.1.3
```
Comment (knqyf263) [inline .github/workflows/bench.yml:10]: We don't want to update minor versions every time. We can just fix the major version.

```suggestion
      uses: actions/setup-go@v2
```

File: .github/workflows/bench.yml [hunk 3]
```diff
@@ -7,13 +7,13 @@ jobs:
...
       with:
         go-version: 1.16
       id: go
 
     - name: Check out code into the Go module directory
-      uses: actions/checkout@v1
+      uses: actions/checkout@v2.3.4
```
Comment (knqyf263) [inline .github/workflows/bench.yml:16]:
```suggestion
      uses: actions/checkout@v2
```

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -10,13 +10,13 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - name: Set up Go
-      uses: actions/setup-go@v1
+      uses: actions/setup-go@v2.1.3
```
Comment (knqyf263) [inline .github/workflows/test.yml:13]:
```suggestion
      uses: actions/setup-go@v2
```

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -10,13 +10,13 @@ jobs:
...
       with:
         go-version: 1.16
       id: go
 
     - name: Check out code into the Go module directory
-      uses: actions/checkout@v1 
+      uses: actions/checkout@v2.3.4
```
Comment (knqyf263) [inline .github/workflows/test.yml:19]:
```suggestion
      uses: actions/checkout@v2
```

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -29,13 +29,13 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - name: Set up Go
-      uses: actions/setup-go@v1
+      uses: actions/setup-go@v2.1.3
```
Comment (knqyf263) [inline .github/workflows/test.yml:32]:
```suggestion
      uses: actions/setup-go@v2
```

File: .github/workflows/test.yml [hunk 7]
```diff
@@ -29,13 +29,13 @@ jobs:
...
       with:
         go-version: 1.16
       id: go
 
     - name: Check out code into the Go module directory
-      uses: actions/checkout@v1
+      uses: actions/checkout@v2.3.4
```
Comment (knqyf263) [inline .github/workflows/test.yml:38]:
```suggestion
      uses: actions/checkout@v2
```

**Review by knqyf263 — CHANGES_REQUESTED** [review]

**Comment by knqyf263** [comment]
Also, the test is failing.

File: .github/dependabot.yml (hunk 1 above)
Comment (mmorel-35) [reply .github/dependabot.yml:6]: What do you mean by scanning, does it includes upgrades ?

**Comment by mmorel-35** [comment]
> Also, the test is failing.
I updated them

File: .github/dependabot.yml (hunk 1 above)
Comment (mmorel-35) [reply .github/dependabot.yml:6]:
```suggestion
```

**Pull request closed without merging**